### PR TITLE
sp-api: Support expanding the macro code

### PR DESCRIPTION
**Pull request by bkchr**
This pr introduces the `expander` crate to expand the generated source code into a file. This gives better error reporting when trying to fix issues in the macro itself as Rustc will point to the line in this file. The feature can be enabled by setting `SP_API_EXPAND=1` at compile time.

Besides that the code generation is changed to fix warnings in the expanded version.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2505221

**Review by melekes — APPROVED** [review]
👍

File: primitives/api/proc-macro/Cargo.toml [hunk 1]
```diff
@@ -21,8 +21,10 @@ syn = { version = "1.0.98", features = ["full", "fold", "extra-traits", "visit"]
 proc-macro2 = "1.0.37"
 blake2 = { version = "0.10.4", default-features = false }
 proc-macro-crate = "1.1.3"
+expander = "1.0.0"
+Inflector = "0.11.4"
```
Comment (melekes) [inline primitives/api/proc-macro/Cargo.toml:25]:
```suggestion
inflector = "0.11.4"
```

Comment (bkchr) [reply primitives/api/proc-macro/Cargo.toml:25]: https://crates.io/crates/Inflector I don't know if cargo is case insensitive :D

Comment (bkchr) [reply primitives/api/proc-macro/Cargo.toml:25]: Just checked, cargo is case sensitive.

**Review by michalkucharczyk — APPROVED** [review]
nice :)

**Review by altonen — APPROVED** [review]

**Pull request merged**